atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -146,7 +145,8 @@ private void waitForRetry() {
         try {
             Thread.sleep(persistentLockRetryWaitMillis);
         } catch (InterruptedException e) {
-            throw Throwables.propagate(e);
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
         }
```

Comment:
maybe there should be a baseline rule for 'catch InterruptedException and immediately rethrow'

Reply:
Would be difficult to code — the internal version of Throwables in atlasdb-commons handles InterruptedException correctly, we wouldn't want to mess up that code.

Reply:
its probably worth an audit of the deprecated `com.google.common.base.Throwables#propagate` as there are definitely cases where we're not resetting interrupts, as well as other places catching `InterruptedException`  (e.g. where we just log or continue in Scrubber and LockAcquirer) that should also likely be resetting.